src/bit-systems/text.ts

```diff
@@ -0,0 +1,31 @@
+import { defineQuery } from "bitecs";
+import { Text as TroikaText } from "troika-three-text";
+import { HubsWorld } from "../app";
+import { TextTag } from "../bit-components";
+
+const textQuery = defineQuery([TextTag]);
+
+export function textSystem(world: HubsWorld) {
+  textQuery(world).forEach(eid => {
+    const text = world.eid2obj.get(eid)! as TroikaText;
+
+    // sync() invokes async text processing in workers.
+    // https://github.com/protectwise/troika/tree/main/packages/troika-three-text#handling-asynchronous-updates
+    //
+    // It is safe to call sync() every frame from the
+    // performance and efficiency perspective because
+    // sync() checks whether to invoke costly processing
+    // inside.
+    //
+    // sync() call can happen one frame after that text
```

Comment:
We choose when to run this system, I think we should just enforce this guarantee.